lib/classification_lifecycle.rb

```diff
@@ -17,20 +17,19 @@ def queue(action)
   end
 
   def transact!(&block)
+    dequeue_for_non_logged_in
     Classification.transaction do
       update_classification_data
-
       #NOTE: ensure the block is evaluated before updating the seen subjects
       # as the count worker won't fire if the seens are set, see #should_count_towards_retirement
       instance_eval(&block) if block_given?
-
       create_recent
       update_seen_subjects
-      #to avoid duplicates in queue, do not refresh the queue before updating seen subjects
-      refresh_queue
       publish_to_kafka
       save_to_cassandra unless Rails.env == 'production'
     end
+    #to avoid duplicates in queue, do not refresh the queue before updating seen subjects
+    refresh_queue
```

Comment:
This should still be within the transaction right? Otherwise it'll update the queue without the seen subjects?